litho-widget/src/main/java/com/facebook/litho/widget/TextSpec.java

```diff
@@ -906,7 +908,8 @@ public void textOffsetOnTouch(int textOffset) {
         highlightStartOffset,
         highlightEndOffset,
         clickableSpanExpandedOffset,
-        c.getLogTag());
+        c.getLogTag(),
+        isRtl(c.getAndroidContext()));
```

Comment:
The layout direction can be explicitly set as well, and should take precedence over this. You can get the resolved layout direction in `@OnBoundsDefined` from the `ComponentLayout` by invoking `layout.getResolvedLayoutDirection()`. The resolved direction can be returned as an `Output` from `@OnBoundsDefined`.

The default value is `YogaDirection.INHERIT`, so here it would be something like:

```
layout == YogaDirection.INHERIT ? isRtl(c.getAndroidContext()) : layout == YogaDirection.RTL
```